### PR TITLE
Build: Add docs & tools for rhsm registration 

### PR DESCRIPTION
**Pull request by jlsherrill**
## Summary
Adds documentation and configuration changes needed to register a client system

## Testing steps

follow directions and see if you can register a system to a content template

## Checklist

- [ ] Tested with snapshotting feature disabled and pulp server URL not configured if appropriate

### CODE REVIEW

File: docs/register_client.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+2. 
+```
+go run cmd/external-repos/main.go nightly-jobs
+```
+3.  Let those repositories snapshot
+
+4.  Note that the current instructions do not work with content_guards enabled.  If `custom_repo_content_guards` is set to true in config.yaml, content access will be denied for custom repositories.
+
+## Configure firewall
+ You may need to adjust your firewall to allow access.  If you run into connection issues, you can set your firewall to be trusted:
```
Comment (swadeley) [inline docs/register_client.md:15]:
```suggestion
 You might need to adjust your firewall to allow access.  If you run into connection issues, you can set your default firewall zone to "trusted":
```

File: docs/register_client.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+subscription-manager register
+```
+Username: admin
+Password: admin
+6. Fetch client UUID, within the VM:
+```bash
+$ subscription-manager identity
+system identity: 97d9b21f-9b49-4eae-ade7-beb2b050dee1
+```
+7.  Add custom repositories if desired via UI or API
```
Comment (swadeley) [inline docs/register_client.md:54]:
```suggestion
7. Add custom repositories if desired via UI or API
```

File: compose_files/pulp/docker-compose.yml [hunk 3]
```diff
@@ -72,12 +72,14 @@ services:
...
       PULP_API_ROOT: /api/pulp/
+      PULP_CONTENT_ORIGIN: "http://pulp.content:8080/"
+      PULP_CONTENT_PATH_PREFIX: "/pulp/content/"
     restart: always
 
   pulp_content:
     image: "quay.io/cloudservices/pulp-ubi:latest"
     deploy:
-      replicas: 1
+      replicas: 2
```
Comment (jlsherrill) [inline compose_files/pulp/docker-compose.yml:82]: meant to change this on pulp_worker, will fix!

File: docs/register_client.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+```bash
+subscription-manager config --server.hostname=pulp.content --server.port=8444  --server.prefix=/candlepin --server.insecure=1
+```
+5. Register the client within the VM:
+```bash
+subscription-manager register
+```
+Username: admin
+Password: admin
+6. Fetch client UUID, within the VM:
```
Comment (rverdile) [inline docs/register_client.md:49]: step 6 is also not on a new line

File: docs/register_client.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+```
+3.  Let those repositories snapshot
+
+4.  Note that the current instructions do not work with content_guards enabled.  If `custom_repo_content_guards` is set to true in config.yaml, content access will be denied for custom repositories.
+
+## Configure firewall
+ You might need to adjust your firewall to allow access.  If you run into connection issues, you can set your default firewall zone to "trusted":
+```bash
+sudo firewall-cmd --set-default-zone=trusted
+```
```
Comment (rverdile) [inline docs/register_client.md:18]: can you also mention this command is for linux? and is there a more specific explanation about what to allow access to? mainly for the non-linux people who would have to do this differently.

**Review by rverdile — APPROVED** [review]

**Pull request merged**